features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/ApiSecurity.jsx

```diff
@@ -0,0 +1,495 @@
+            !tempSecurityScheme.includes(securitySchemaValues.basicAuth);
+
+        if (!mutualSSLWithOauthOrBasicAuth) {
+            if (mutualSSLOnly) {
+                addToArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+                removeFromArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+            } else {
+                addToArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+                removeFromArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+            }
+        }
+
+        setSecurityScheme(tempSecurityScheme);
+    };
+
+    const handleAPISecurityMandatoryStateChange = apiSecurityScheme => (event) => {
+        const { value, name } = event.target;
+        let tempSecurityScheme = securityScheme;
+        if (!securityScheme) tempSecurityScheme = apiSecurityScheme;
+        const isMandatory = value === 'Mandatory';
+
+        if (name === 'mutualSSLMandatoryState') {
+            if (isMandatory) addToArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+            else {
+                removeFromArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+                if (tempSecurityScheme.includes(securitySchemaValues.oauth2) ||
+                tempSecurityScheme.includes(securitySchemaValues.basicAuth)
+                ) {
+                    addToArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+                }
+            }
+        } else if (name === 'oauthBasicAuthMandatoryState') {
+            if (isMandatory) addToArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+            else {
+                removeFromArray(securitySchemaValues.oauthBasicAuthMandatory, tempSecurityScheme);
+                if (tempSecurityScheme.includes(securitySchemaValues.mutualSSL)) {
+                    addToArray(securitySchemaValues.mutualSSLMandatory, tempSecurityScheme);
+                }
+            }
+        }
+        setSecurityScheme(tempSecurityScheme);
+    };
+
+    const showMandatoryOption = (apiSecurityScheme) => {
+        let tempSecurityScheme = securityScheme;
+        if (!tempSecurityScheme) tempSecurityScheme = apiSecurityScheme;
+
+        const mutualSSLWithOauthOrBasicAuth = tempSecurityScheme.includes(securitySchemaValues.mutualSSL) &&
+        (tempSecurityScheme.includes(securitySchemaValues.oauth2) ||
+        tempSecurityScheme.includes(securitySchemaValues.basicAuth));
+
+        return mutualSSLWithOauthOrBasicAuth;
+    };
+
+    const getAPISecurityMessage = (apiSecurityScheme) => {
+        let tempSecurityScheme = securityScheme;
+        if (!tempSecurityScheme) tempSecurityScheme = apiSecurityScheme;
+        if (tempSecurityScheme.includes(securitySchemaValues.mutualSSLMandatory) &&
+        tempSecurityScheme.includes(securitySchemaValues.oauthBasicAuthMandatory)) {
```

Comment:
shouldn't the second line of the if condition be left indented?

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/commit/07943fe71a315272620023b9be9f3fc4d43d1f4e